### PR TITLE
Handle multiple chains in BlockchainRegistry

### PR DESCRIPTION
**Pull request by bbondy**
- Makes the BlockchainRegistry functions take in a chain Id. 
- Parses 2 new token lists for which we don't have files for yet (but we will)
- Adds a coingecko_id member to BlockchainToken.  Front end code should use this for price data first if it's available.

Resolves https://github.com/brave/brave-browser/issues/20344
Resolves https://github.com/brave/brave-browser/issues/20342

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 1]
```diff
@@ -715,7 +716,8 @@ void BraveWalletService::AddSuggestTokenRequest(
                    request->token->is_erc721, chain_id);
 
   if (!token && chain_id == mojom::kMainnetChainId)
-    token = BlockchainRegistry::GetInstance()->GetTokenByContract(addr);
+    token = BlockchainRegistry::GetInstance()->GetTokenByContract(
+        mojom::kMainnetChainId, addr);
```
Comment (yrliou) [inline components/brave_wallet/browser/brave_wallet_service.cc:720]: Probably not within this PR's scope, now that we'll have registry data for other chains, we should query the registry using chain_id for other chains too and use it if found.

**Review by yrliou — APPROVED** [review]
++

File: components/brave_wallet/browser/brave_wallet_service.cc (hunk 1 above)
Comment (bbondy) [reply components/brave_wallet/browser/brave_wallet_service.cc:720]: noted, thx

Comment (bbondy) [reply components/brave_wallet/browser/brave_wallet_service.cc]: Fixed in rebase here https://github.com/brave/brave-core/pull/11791/commits/60036d25b2dac7b15ee980dabb1b1bd366d447fb

**Review by yrliou — APPROVED** [review]
Thanks for addressing multi-chain registry support for`AddSuggestToken`.

**Pull request merged**